### ISSUE TITLE
LearningRateMonitor may not require loggers

### ISSUE THREAD

**Issue by dkajtoch** [PyTorchLightning/pytorch-lightning#17159]
### Description & Motivation

I was lately been working with lightning and mlflow autolog feature (I couldn't use lightning's mlflow logger since I mlflow context manager does more stuff and allows to log outside loops) and wanted to log metrics from the `LearningRateMonitor`. Unfortunately, since I do not attach any loggers (I do not want default Tensorboard) it is impossible to use this callback due to assert in the code. My question is why can't we use `pl_module.log_dict(...)` from the callback instead of `logger.log_metrics`? With `pl_module.log` you can specify whether you want the metric to land in loggers or not. For me, it is more general. Although I am not sure if proper step value will be used. 

### Pitch

I want the `LearningRateMonitor` to use `pl_module.log_dict(...)` in `on_train_batch_start` and `on_train_epoch_start` which will not require the presence of loggers and external tools like `mlflow.pytorch.autolog` will be able to log metrics. 

```diff
- for logger in trainer.loggers:
-    logger.log_metrics(latest_stat, step=trainer.fit_loop.epoch_loop._batches_that_stepped)
+ pl_module.log_dict(latest_stat, on_step=True, on_epoch=False)
```
A private attribute is necessary in order to correctly log this metric.

```diff
- if not trainer.loggers:
- raise MisconfigurationException(
-     "Cannot use `LearningRateMonitor` callback with `Trainer` that has no logger."
- )
```

cc @borda

**Comment by shenoynikhil** [comment]
I agree. This is also a hastle when debugging without loggers. I need to remove this callback specifically because it doesn't work when I turn off loggers.
